### ISSUE TITLE
tfjs 2.8.1 change in behavior in tf.reshape causes errors

### ISSUE THREAD

**Issue by vladmandic** [tensorflow/tfjs#4429]
upon upgrade from 2.7.0 to 2.8.1, my existing app fails with:
```
Uncaught (in promise) Error: The implicit shape can't be a fractional number. Got 266 / 3
    at Object.inferFromImplicitShape (util_base.ts:328)
    at Object.reshape76 [as kernelFunc] (Reshape.ts:35)
    at kernelFunc3 (engine.ts:590)
    at engine.ts:660
    at Engine.scopedRun (engine.ts:453)
    at Engine.runKernelFunc (engine.ts:657)
    at Engine.runKernel (engine.ts:522)
    at reshape_ (reshape.ts:60)
    at reshape__op (operation.ts:51)
```

this may be desired behavior, but it breaks several existing models published on `tfhub`!

e.g, i'm using `blazeface` where `reshape` is failing on results of model inference, so not like that can be changed easily:
```js
const [contours, confidence, contourCoords] = this.meshDetector.predict(face);
const coordsReshaped = tf.reshape(contourCoords, [-1, 3]);
let rawCoords = coordsReshaped.arraySync();
```

**Comment by lina128** [comment]
Which backend has this behavior?

**Comment by vladmandic** [comment]
webgl backend

**Comment by scottzockoll** [comment]
@lina128 I am also receiving this error after upgrading to 2.8.1 with webgl.

**Comment by p-dimi** [comment]
Same thing here!

**Comment by designerzen** [comment]
Same here! When trying to fetch face landmarks :

` const predictions = await model.estimateFaces({ input: video }) `

Browser returns 

`util_base.ts:328 Uncaught (in promise) Error: The implicit shape can't be a fractional number. Got 266 / 3`

**Comment by rthadur** [comment]
cc @pyu10055 @annxingyuan @tafsiri

**Comment by dewball345** [comment]
Yup, I'm getting the same error as well! I still don't understand why demos like these https://mediapipe-facemesh-3js-tf2.glitch.me/ work. I tried changing the version of my facemesh model but that doesn't do anything

**Comment by vladmandic** [comment]
@dewball345 because it's hard coded to load tfjs 2.4.0. anything up to 2.7.0 works fine, 2.8.0 and 2.8.1 don't.

**Comment by dewball345** [comment]
yeah i realized that now. A temporary fix would be to just use the older version of tensorflow?

**Comment by vladmandic** [comment]
I'm staying on 2.7.0 until this is resolved.

**Comment by annxingyuan** [comment]
Hi @lina128 - I ran a git bisect and it seems to be traceable to this PR: https://github.com/tensorflow/tfjs/pull/4187

**Comment by lina128** [comment]
Thanks for the bug report, we will release the fix in 2.8.2 next week.
